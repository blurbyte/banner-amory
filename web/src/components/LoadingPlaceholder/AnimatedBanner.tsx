// background banner in animation

import * as React from 'react';
import styled from 'styled-components';
import { transparentize } from 'polished';

import { Color } from '@styles/color';
import { slideIn } from '@styles/animations';
import { Banner } from '../Icons';

const Icon = styled.div`
  color: ${transparentize(0.9, Color.Blue)};
  opacity: 0;
  animation: ${slideIn(-12)} 400ms ease-in forwards;
`;

class BannerTransition extends React.Component {
  render() {
    return (
      <Icon>
        <Banner />
      </Icon>
    );
  }
}

export default BannerTransition;
