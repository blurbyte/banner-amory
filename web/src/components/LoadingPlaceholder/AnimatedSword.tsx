// sword in animation

import * as React from 'react';
import styled, { keyframes } from 'styled-components';

import { slideIn } from '@styles/animations';
import { Sword } from '../Icons';

const moveSwordFill = keyframes`
  0% {
    y: 54;
  }
  10% {
    y: 54;
  }
  40% {
    y: 250;
  }
  60% {
    y: 250;
  }
  90% {
    y: 54;
  }
  100% {
    y: 54;
  }
`;

const Icon = styled.div`
  position: absolute;
  top: 3rem;
  left: 2.5rem;
  opacity: 0;
  animation: ${slideIn(-4)} 600ms cubic-bezier(0.175, 0.885, 0.32, 1.6) forwards 200ms;

  #swordFill {
    y: 54;
    animation: ${moveSwordFill} 12000ms ease-in infinite;
  }
`;

class AnimatedSword extends React.Component {
  render() {
    return (
      <Icon>
        <Sword />
      </Icon>
    );
  }
}

export default AnimatedSword;
