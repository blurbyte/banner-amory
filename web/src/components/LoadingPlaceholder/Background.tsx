import styled from 'styled-components';

import { Color } from '@styles/color';

const Background = styled.div`
  position: absolute;
  z-index: 999;
  top: 0;
  left: 0;
  width: 100%;
  min-height: 29rem;
  height: 100%;
  overflow: hidden;
  background: ${Color.White};
  display: flex;
  justify-content: center;
`;

export default Background;
