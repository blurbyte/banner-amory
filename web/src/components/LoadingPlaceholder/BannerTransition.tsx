// background banner in and out animation

import * as React from 'react';
import styled from 'styled-components';
import { CSSTransition } from 'react-transition-group';
import { transparentize } from 'polished';

import { Color } from '@styles/color';
import { Banner } from '../Icons';

const Icon = styled.div`
  color: ${transparentize(0.9, Color.Blue)};
`;

const Transition = styled(CSSTransition)`
  &.banner-enter {
    opacity: 0;
    transform: translateY(-12rem);
  }

  &.banner-enter-active {
    opacity: 1;
    transform: translateY(0);
    transition: opacity 400ms ease-in, transform 600ms ease-in;
  }

  &.banner-exit {
    opacity: 1;
    transform: translateY(0);
  }

  &.banner-exit-active {
    opacity: 0;
    transform: translateY(-12rem);
    transition: opacity 400ms ease-out 400ms, transform 400ms ease-out 400ms;
  }
`;

type BannerTransitionProps = {
  in: boolean;
};

class BannerTransition extends React.Component<BannerTransitionProps> {
  render() {
    return (
      <Transition
        in={this.props.in}
        classNames="banner"
        timeout={600}
        appear
        mountOnEnter
        unmountOnExit
      >
        <Icon>
          <Banner />
        </Icon>
      </Transition>
    );
  }
}

export default BannerTransition;
