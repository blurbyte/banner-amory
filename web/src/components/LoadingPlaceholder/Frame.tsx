// banner and sword wrapper

import styled from 'styled-components';

const Frame = styled.div`
  position: relative;
  width: 15rem;
`;

export default Frame;
