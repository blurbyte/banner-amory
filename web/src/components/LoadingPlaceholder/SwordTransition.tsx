// sword in and out animation

import * as React from 'react';
import styled, { keyframes } from 'styled-components';
import { CSSTransition } from 'react-transition-group';

import { Sword } from '../Icons';

const moveSwordFill = keyframes`
  0% {
    y: 54;
  }
  10% {
    y: 54;
  }
  40% {
    y: 250;
  }
  60% {
    y: 250;
  }
  90% {
    y: 54;
  }
  100% {
    y: 54;
  }
`;

const Icon = styled.div`
  position: absolute;
  top: 3rem;
  left: 2.5rem;

  #swordFill {
    y: 54;
    animation: ${moveSwordFill} 12000ms ease-in infinite;
  }
`;

const Transition = styled(CSSTransition)`
  &.sword-enter {
    opacity: 0;
    transform: translateY(-4rem);
  }

  &.sword-enter-active {
    opacity: 1;
    transform: translateY(0);
    transition: opacity 200ms ease-in 200ms,
      transform 400ms cubic-bezier(0.175, 0.885, 0.32, 1.6) 200ms;
  }

  &.sword-exit {
    opacity: 1;
    transform: translateY(0);
  }

  &.sword-exit-active {
    opacity: 0;
    transform: translateY(-4rem);
    transition: opacity 400ms ease-out, transform 400ms cubic-bezier(0.6, -0.6, 0.735, 0.045);
  }
`;

type SwordTransitionProps = {
  in: boolean;
};

class SwordTransition extends React.Component<SwordTransitionProps> {
  render() {
    return (
      <Transition
        in={this.props.in}
        classNames="sword"
        timeout={600}
        appear
        mountOnEnter
        unmountOnExit
      >
        <Icon>
          <Sword />
        </Icon>
      </Transition>
    );
  }
}

export default SwordTransition;
