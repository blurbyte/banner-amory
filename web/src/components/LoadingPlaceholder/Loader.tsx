import * as React from 'react';

import Background from './Background';
import Frame from './Frame';
import Banner from './AnimatedBanner';
import Sword from './AnimatedSword';

class Loader extends React.Component {
  render() {
    return (
      <Background>
        <Frame>
          <Banner />
          <Sword />
        </Frame>
      </Background>
    );
  }
}

export default Loader;
