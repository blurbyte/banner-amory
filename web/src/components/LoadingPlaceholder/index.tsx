// custom loading placeholder
// display loader after delay
// so if data got loaded fast it won't be shown

import * as React from 'react';
import Loader from './Loader';

type LoadingPlaceholderState = Readonly<{
  showLoader: boolean;
}>;

class LoadingPlaceholder extends React.Component<{}, LoadingPlaceholderState> {
  readonly state = {
    showLoader: false
  };

  private delay = setTimeout(() => {
    this.setState({ showLoader: true });
  }, 1000);

  componentWillUnmount() {
    clearTimeout(this.delay);
  }

  render() {
    return this.state.showLoader ? <Loader /> : null;
  }
}

export default LoadingPlaceholder;
