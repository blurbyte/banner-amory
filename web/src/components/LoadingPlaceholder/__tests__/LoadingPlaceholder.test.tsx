import '../../../__mocks__/ShallowReactTransitionGroup.mock';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import LoadingPlaceholder from '../index';

test('renders correctly when visible', () => {
  const component = renderer.create(<LoadingPlaceholder isVisible />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('renders correctly when hidden', () => {
  const component = renderer.create(<LoadingPlaceholder isVisible={false} />);
  expect(component.toJSON()).toMatchSnapshot();
});
