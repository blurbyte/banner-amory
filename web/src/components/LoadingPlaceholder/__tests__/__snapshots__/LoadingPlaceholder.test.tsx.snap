// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when hidden 1`] = `
<div
  className="sc-bdVaJa iRgTQi"
>
  <div
    className="sc-bwzfXH jqVWXS"
  >
    CssTransition
    CssTransition
  </div>
</div>
`;

exports[`renders correctly when visible 1`] = `
<div
  className="sc-bdVaJa iRgTQi"
>
  <div
    className="sc-bwzfXH jqVWXS"
  >
    CssTransition
    CssTransition
  </div>
</div>
`;
